src/app.ts

```diff
@@ -1,43 +1,64 @@
-import {
-  ChainName,
-  HyperlaneApp,
-  objMap,
-  promiseObjAll,
-} from '@hyperlane-xyz/sdk';
+import { BigNumberish } from 'ethers';
 
-import { HypERC20Contracts, HypERC721Contracts } from './contracts';
+import { ChainName, GasRouterApp, RouterContracts } from '@hyperlane-xyz/sdk';
+import { Address } from '@hyperlane-xyz/utils/dist/src/types';
 
-export class HypERC20App<Chain extends ChainName> extends HyperlaneApp<
-  HypERC20Contracts,
-  Chain
-> {
-  getSecurityModules = () =>
-    promiseObjAll(
-      objMap(this.contractsMap, (_, contracts) =>
-        contracts.router.interchainSecurityModule(),
-      ),
-    );
+import { HypERC20Contracts, HypERC721Contracts } from './contracts';
+import { TokenRouter } from './types';
 
-  getOwners = () =>
-    promiseObjAll(
-      objMap(this.contractsMap, (_, contracts) => contracts.router.owner()),
+class HyperlaneTokenApp<
+  Contracts extends RouterContracts<TokenRouter>,
+> extends GasRouterApp<Contracts> {
+  async transfer(
+    origin: ChainName,
+    destination: ChainName,
+    recipient: Address,
+    amountOrId: BigNumberish,
+  ) {
+    const originRouter = this.getContracts(origin).router;
+    const destinationDomain = this.multiProvider.getDomainId(destination);
+    const gasPayment = await originRouter.quoteGasPayment(destinationDomain);
```

Comment:
This technically won't work on the audited version of warp routes. Dont have to fix it here, but just flagging